lib/stdlib/doc/src/lists.xml

```diff
@@ -196,6 +200,10 @@
         <c>I</c> is the position of <c>H</c> in
         <c><anno>List1</anno></c>. The enumeration starts with
         <c><anno>Index</anno></c> and increases by 1 in each step.</p>
+        <p>The following equalities hold for all enumerations:</p>
+        <code type="none">
+length(lists:enumerate(I, List)) =:= length(List)
+lists:enumerate(I, List) =:= element(1, lists:mapfoldl(fun(T,Acc) -> {{Acc,T}, Acc+1} end, I, List))</code>
```

Comment:
```suggestion
        <p>That is, <c>enumerate/2</c> behaves as if it had been defined as follows:</p>
        <code type="erl">
enumerate(I, List) ->
  {List1, _ } = lists:mapfoldl(fun(T, Acc) -> {{Acc, T}, Acc+1} end, I, List),
  List1.</code>
```

`lists:filtermap/2` uses the this style to describe what the implementation does, so I suggest that we mimic it.

Reply:
I got my inspiration from `lists:seq/2`. Also, I think if we are giving a definition example, why not relate to `zip` and `seq`? It is more explicit to my eyes than `mapfoldl`.

Reply:
Yes, good idea. Use lists:seq + lists:zip as an example instead.

Having a definition example helps when someone comes across this and needs it but with a small variation. They can then just copy the example and adapt it to their needs.

Reply:
The idea with using lists:mapfold/3 was that you could change the fun to add a third element,  
or step 10 in each increment,  lists:mapfoldl can do arbitrary change that lists:seq and lists:zip can't do.

Reply:
Check it out

Reply:
I updated it to use zip&seq, please check it out

Reply:
Ok, it is smaller but that also shows how to write the function in the worst way, from a performance perspective.

The definition is pretty clear from the usage example, so we have already decided that we want the
mapfold example to avoid other variants of this function.

```
<code>
enumerate(I, List) ->
  {List1, _} = lists:mapfoldl(fun(T, Index) -> {{Index, T}, Index+1} end, I, List),
  List1.
</code>

```